### PR TITLE
bug_fix:修正移动镜头将资源点置于屏幕中央后陷入循环的问题（#1954）

### PR DESCRIPTION
**Pull request by piliapi**
修正移动镜头将资源点置于屏幕中央后，获得道具时陷入循环的问题

### CODE REVIEW

**Comment by LmeSzinc** [comment]
需要修复的是 `area_cross_area` 函数
https://github.com/LmeSzinc/AzurLaneAutoScript/blob/43dfe6ef6cf86714dbcedd44871476f9eaa4cbbf/module/base/utils.py#L315-L329

**Comment by piliapi** [comment]
尝试改了下，初步测试好像没问题了
‘return point_in_area((area1[0], area1[1]), area2, threshold=threshold) and ~point_in_area((area1[2], area1[3]), area2, threshold=threshold)\
           or point_in_area((area1[2], area1[1]), area2, threshold=threshold) and ~point_in_area((area1[0], area1[3]), area2, threshold=threshold)\
           or point_in_area((area1[0], area1[3]), area2, threshold=threshold) and ~point_in_area((area1[2], area1[1]), area2, threshold=threshold)\
           or point_in_area((area1[2], area1[3]), area2, threshold=threshold) and ~point_in_area((area1[0], area1[1]), area2, threshold=threshold) ’

**Comment by piliapi** [comment]
> 需要修复的是 `area_cross_area` 函数，判断两个矩形是否相交，代码里写得不对
> 
> https://github.com/LmeSzinc/AzurLaneAutoScript/blob/43dfe6ef6cf86714dbcedd44871476f9eaa4cbbf/module/base/utils.py#L315-L329
尝试改了下，初步测试没出现循环点击的问题了
'return point_in_area((area1[0], area1[1]), area2, threshold=threshold) and ~point_in_area((area1[2], area1[3]), area2, threshold=threshold)\
           or point_in_area((area1[2], area1[1]), area2, threshold=threshold) and ~point_in_area((area1[0], area1[3]), area2, threshold=threshold)\
           or point_in_area((area1[0], area1[3]), area2, threshold=threshold) and ~point_in_area((area1[2], area1[1]), area2, threshold=threshold)\
           or point_in_area((area1[2], area1[3]), area2, threshold=threshold) and ~point_in_area((area1[0], area1[1]), area2, threshold=threshold) '

File: module/base/utils.py [hunk 1]
```diff
@@ -323,10 +323,10 @@ def area_cross_area(area1, area2, threshold=5):
...
         bool:
     """
-    return point_in_area((area1[0], area1[1]), area2, threshold=threshold) \
-           or point_in_area((area1[2], area1[1]), area2, threshold=threshold) \
-           or point_in_area((area1[0], area1[3]), area2, threshold=threshold) \
-           or point_in_area((area1[2], area1[3]), area2, threshold=threshold)
+    return point_in_area((area1[0], area1[1]), area2, threshold=threshold) and ~point_in_area((area1[2], area1[3]), area2, threshold=threshold)\
+           or point_in_area((area1[2], area1[1]), area2, threshold=threshold) and ~point_in_area((area1[0], area1[3]), area2, threshold=threshold)\
+           or point_in_area((area1[0], area1[3]), area2, threshold=threshold) and ~point_in_area((area1[2], area1[1]), area2, threshold=threshold)\
+           or point_in_area((area1[2], area1[3]), area2, threshold=threshold) and ~point_in_area((area1[0], area1[1]), area2, threshold=threshold)
```
Comment (LmeSzinc) [inline module/base/utils.py:329]: 判断两个矩形是否相交 这个问题有现成的算法的，可以搜一下，按点是否在矩形内不能涵盖所有情况

File: module/handler/mystery.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 class MysteryHandler(StrategyHandler, EnemySearchingHandler):
     _get_ammo_log_timer = Timer(3)
     carrier_count = 0
-
+
```
Comment (LmeSzinc) [inline module/handler/mystery.py:13]: 回退多余的 空格

File: module/base/utils.py (hunk 1 above)
Comment (Horizon101011) [reply module/base/utils.py:329]: 给出一个参考的实现
```python
def area_cross_area(area1, area2, threshold=5):
	central_1_x = (area1[0] + area1[2]) // 2
	central_2_x = (area2[0] + area2[2]) // 2
	central_1_y = (area1[1] + area1[3]) // 2
	central_2_y = (area2[1] + area2[3]) // 2
	
	r_1_x = abs(area1[0] - area1[2])
	r_2_x = abs(area2[0] - area2[2])
	r_1_y = abs(area1[1] - area1[3])
	r_2_y = abs(area2[1] - area2[3])
	
	dx = abs(central_1_x - central_2_x)
	dy = abs(central_1_y - central_2_y)
	
	return abs(r_1_x - r_2_x) - threshold < dx < r_1_x + r_2_x + threshold \
		and abs(r_1_y - r_2_y) - threshold < dy < r_1_y + r_2_y + threshold
```

Comment (hgjazhgj) [reply module/base/utils.py:329]:
```python3
def a_c_a(area1,area2,threshold):
    return max(area1[0],area2[0])-threshold<min(area1[2],area2[2])+threshold \
       and max(area1[1],area2[1])-threshold<min(area1[3],area2[3])+threshold
```
存在两个矩形的相交矩形

Comment (hgjazhgj) [reply module/base/utils.py:329]:
```python3
def a_c_a(area1,area2,threshold):
    return area1[2]+threshold>area2[0]-threshold and area1[3]+threshold>area2[1]-threshold \
       and area2[2]+threshold>area1[0]-threshold and area2[3]+threshold>area1[1]-threshold
```
not(一个在另一个的左边或上边)

**Pull request closed without merging**